src/test/java/com/exasol/adapter/dialects/rls/RowLevelSecurityQueryRewriterTest.java

```diff
@@ -0,0 +1,148 @@
+package com.exasol.adapter.dialects.rls;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.*;
+
+import java.math.BigDecimal;
+import java.sql.*;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import com.exasol.ExaMetadata;
+import com.exasol.adapter.AdapterException;
+import com.exasol.adapter.AdapterProperties;
+import com.exasol.adapter.dialects.AbstractQueryRewriterTestBase;
+import com.exasol.adapter.dialects.SqlDialect;
+import com.exasol.adapter.dialects.exasol.ExasolMetadataReader;
+import com.exasol.adapter.jdbc.RemoteMetadataReader;
+import com.exasol.adapter.metadata.ColumnMetadata;
+import com.exasol.adapter.metadata.DataType;
+import com.exasol.adapter.metadata.DataType.ExaCharset;
+import com.exasol.adapter.metadata.TableMetadata;
+import com.exasol.adapter.sql.*;
+
+@ExtendWith(MockitoExtension.class)
+class RowLevelSecurityQueryRewriterTest extends AbstractQueryRewriterTestBase {
+    private SqlDialect dialect;
+    private AdapterProperties properties;
+    private RemoteMetadataReader metadataReader;
+    @Mock
+    private Connection connectionMock;
+    @Mock
+    private DatabaseMetaData databaseMetaDataMock;
+
+    @BeforeEach
+    void beforeEach() throws SQLException {
+        this.exaMetadata = mock(ExaMetadata.class);
+        this.rawProperties = new HashMap<>();
+        this.dialect = new RowLevelSecurityDialect(null, AdapterProperties.emptyProperties());
```

Comment:
Why initialize this if you use `AdapterProperties.emptyProperties()` in the next line?

Reply:
Because I need adapterProperties in metadataReader. Moved the initialization down

Reply:
done